server/controller/product.js

```diff
@@ -61,7 +61,7 @@ const updateProduct = async (req, res) => {
   }
 };
 
-// Delete a product 
+// Delete a product
 const deleteProduct = async (req, res) => {
```

Comment:
There's a huge bug here. You're first deleting the product and then checking for owner and if the product exist? Come on guys please think before you write the code

and also do not chain the if-else unnecessarily, it makes the code look complicated. Try this

```javascript
const deleteProduct = async (req, res) => {
  const { id } = req.params;

  try {
    const product = await Product.findByID(id);

    if (!product) {
        return res.status(404).json({ error: 'Product not found' });
    }
    
    if (product.owner.toString() !== req.user.userId) {
       return res.status(403).json({ error: 'Access denied. User does not own the product.' });
    }
    
    await Product.findByIdAndDelete(id);

    return res.status(200).send(true);
      
  } catch (error) {
    return res.status(500).json({ error: 'Error deleting product' });
  }
};

```

Look simple this code looks, compared to the current implementation